File: LEVEL1/비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for i in range(n):
        arr1[i] = list(format(arr1[i], 'b'))
        arr2[i] = list(format(arr2[i], 'b'))
        
        while len(arr1[i]) < n:
            arr1[i].insert(0, 0)
        while len(arr2[i]) < n:
            arr2[i].insert(0, 0)
        
        for j in range(n):
            if int(arr1[i][j]) != int(arr2[i][j]):
                arr1[i][j] = 1
        
        a = ''        
        for k in range(n):
            if int(arr1[i][k]) > 0:
                a += '#'
            else:
                a += ' '
        answer.append(a)

    return answer

File: Javascript/KAKAO/비밀지도.js
function solution(n, arr1, arr2) {
  var answer = [];
  for (let i = 0; i < n; i++) {
    let tmp = (arr1[i] | arr2[i]).toString(2).replace(/1/g, '#').replace(/0/g, ' ');
    if (tmp.length !== n) {
      while (n > tmp.length) {
        tmp = ' ' + tmp;
      }
    }
    answer.push(tmp);
  }
  return answer;
}
